contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./IFeralfileSaleData.sol";
+import "./ECDSASigner.sol";
+import "./IFeralfileVault.sol";
+
+contract IFeralfileExhibitionV4 is IFeralfileSaleData {
+    function buyArtworks(
+        bytes32 r_,
+        bytes32 s_,
+        uint8 v_,
+        SaleData calldata saleData_
+    ) external payable {}
+}
+
+contract FeralfileEnglishAuction is Ownable, IFeralfileSaleData, ECDSASigner {
+    struct EnglishAuction {
+        bytes32 id;
+        uint256 startAt;
+        uint256 endAt;
+        uint256 extendDuration;
+        uint256 extendThreshold;
+        uint256 minIncreaseFactor;
+        uint256 minIncreaseAmount;
+        uint256 minPrice;
+        bool isSettled;
+    }
+
+    struct Bid {
+        address bidder;
+        uint256 amount;
+        bool fromFeralFile;
+    }
+
+    struct LatestAuctionStatus {
```

Comment:
I think this structure could be redundant. Instead, we could query the auction info and the highest bid separately, right?